content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+
+* **Frequency weights** are integers that denote how many times a particular row of data has been observed. They help compress redundant rows into a smaller format. 
+* **Importance weights** focus on how much each row of the data set should influence the model fit. These can be based on data or arbitrarily set to achieve some goal. 
+* When survey respondents have different probabilities of selection, (inverse) **probability weights** can help remove bias from the results of a data analysis. 
+* If a data point has an associated precision, **analytic weighting** have help a model focus on the data points with less uncertainty (such as in meta-analysis).
+
+There are undoubtedly more types of weights in other domains. 
+
+Generally speaking, case weights are usually non-negative. 
+
+## How are they used in base R?
+
+The best example is for categorical data where a small number of possible categories are observed many times. For example, the `UCBAdmissions` contains "Aggregate data on applicants to graduate school at Berkeley for the six largest departments in 1973 classified by admission and sex."
+
+```{r UCBAdmissions}
+data("UCBAdmissions")
+UCBAdmissions
+```
+
+This is a 3D array, so let's convert to a rectangular data format:
+
+```{r df}
+library(tidymodels)
+
+ucb <- 
+  as_tibble(UCBAdmissions) %>% 
+  mutate(across(where(is.character), ~ as.factor(.)))
+ucb
+```
+
+There are `r nrow(ucb)` possible configurations of the variables but a total of `r sum(ucb$n)` observations. If we want to model the data in this format, using a logistic regression, we could  use
+
+```{r glm}
+glm_fit <-
+  glm(
+    Admit ~ Gender + Dept,
+    data = ucb,
+    weights = n,
+    family = "binomial"
+  )
+glm_fit
+```
+
+
+_This is not quite right though_. There are 12 combinations of `Gender` and `Dept`. How can the model have 23 total degrees of freedom?
+
+If we are treating our data as binomial, the traditional method for fitting this model is to convert the data to a format with columns for the number of events and non-events (per covariate pattern). Let's convert our data into that format:
+
+```{r glm-events}
+ucb_grouped_data <- 
+  ucb %>% 
+  pivot_wider(
+    id_cols = c(Gender, Dept),
+    names_from = Admit,
+    values_from = n,
+    values_fill = 0L
+  )
+ucb_grouped_data
+```
+
+Now, since there are really only `r nrow(ucb_grouped_data)` covariate combinations, the appropriate model can be created. 
+
+
+```{r glm-grouped}
+glm(
+  cbind(Rejected, Admitted) ~ Gender + Dept,
+  data = ucb_grouped_data,
+  family = binomial
+)
+```
+
+In both cases the model coefficients are the same; the standard errors and degrees of freedom are correct for the model with grouped data though. 
+
+## Why is this so complicated? 
+
+Traditionally, the use of weights in base R functions are to fit the model and to report limited measures of model fit. Here, `glm()` reports the deviance while `lm()` shows estimates of the RMSE and adjusted-R<sup>2</sup>.
```

Comment:
replace base R with classical statistical models

Reply:
Maybe "classical implementations of statistical models" or something like that, since it's about the R implementation, not the model itself... "Traditional implementations of statistical models use weights to fit the model and report limited measures of model fit."